### ISSUE TITLE
Keyboard not being shared since macOS update (10.14.6 (18G87))

### ISSUE THREAD

**Issue by ghost** [input-leap/input-leap#393]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/393
> 
> Issue created on: 2019-08-07 by @domparry
> Issue last updated on: 2020-12-04

### Operating Systems ###

Server: macOS Mojave 10.14.6 (18G87)

Client: macOS High Sierra 10.13.6

### Barrier Version ###

2.1.0 and 2.3.0 

### Steps to reproduce bug ###

1. Open Barrier on both machines
2. Move the mouse into the screen of the client.
3. Mouse control is transferred and works normally.
4. Keyboard still active on server, but not on client

### Other info ###

* When did the problem start to occur? Since running the latest macOS update (on 7-08-2019)
* Is there a way to work around it? No
* Does this bug prevent you from using Barrier entirely? No

---

> Commented on: 2019-08-19 by @domparry

Just as an update, I've tested this on 2.3.1 and it still persists.

---

> Commented on: 2019-08-30 by @domparry

Something I see in the logs when I switch to the other machine:
```
2019-08-30 17:28:53.195 barriers[1353:12766] pid(1353)/euid(501) is calling TIS/TSM in non-main thread environment, ERROR : This is NOT allowed. Please call TIS/TSM in main thread!!!
```

---

> Commented on: 2019-08-30 by @AdrianKoshka

That's a long-time warning, iirc it's doesn't interfere with anything.

---

> Commented on: 2019-09-02 by @metadave

I'm experiencing the same issue. I added a bunch of traces throughout the source to log events (EventQueue, OSXKeyState, OSXScreen etc), and it appears that only mouse events are being fired. If anyone has pointers on where I should be looking in the source, I'd be happy to debug some more.

---

> Commented on: 2019-09-03 by @domparry

Any work around on your side to get it going again @metadave ?

---

> Commented on: 2019-09-05 by @jbirthisel

This is happening to me off and on all the time.

Example: Today quit working in morning. Reset both server and client side. Rebooted client even. No fix. Then 5 minutes ago (14:00) the keyboard just started working again.

---

> Commented on: 2019-09-05 by @metadave

> Any work around on your side to get it going again @metadave ?

not yet, still debugging and learning about iokit. All signs currently point to keyboard events not being fired on the server. 

edit: 
> All signs currently point to keyboard events not being fired on the server. 

control/option/command appear to fire events, letter/number keys do not

---

> Commented on: 2019-09-06 by @noisyshape

I couldn't reproduce this with VMware guests and I'm not seeing any difference in behavior with 10.14.6. This may be identical to https://github.com/symless/synergy-core/issues/6513 where supposedly it's something to do with input source settings.

When all else fails, post full debug logs.

---

> Commented on: 2019-09-08 by @domparry

I've double checked the keyboard input source settings as described in that issue, and that's not the case here. @noisyshape When you say full debug logs, are you referring to something more than the logs shown from the `show log` function? If there's a way to turn on more detailed logs, let me know and I'll happily do so.

---

> Commented on: 2019-09-09 by @noisyshape

@domparry In the settings dialog (F4) there is a logging level option. Set it to Debug2 and restart the client/server. If you can log a short session with a couple of failed keystrokes that could be helpful. You can use pastebin for this.

---

> Commented on: 2019-09-09 by @domparry

*** Edit - I didn't restart barrier after setting the debug level, so re-pasting the logs ***

Here you go @noisyshape . I've included both the client and server logs below.

Client Log: https://pastebin.com/EcCyjHMR
Server Log:https://pastebin.com/1VwreTTK

---

> Commented on: 2019-09-13 by @domparry

Please let me know if there is anything else I can provide to help find the cause of this. It essentially makes barrier unusable. 

---

> Commented on: 2019-09-17 by @metadave

As barrier and synergy aren't working for me because of this bug, I tried using https://www.sharemouse.com/. It has the same problem with keyboard sharing, _however, it also shows an orange icon when running that says "Keyboard not working?". Clicking on the icon takes you to the [ShareMouse support page](https://www.sharemouse.com/doc/troubleshooting/#keyboard) that describes apps that leave [MacOS Secure Input](https://developer.apple.com/library/archive/technotes/tn2150/_index.html) enabled. 

tl;dr I started killing apps until my keyboard worked again. Some possibly useful info [here](https://security.stackexchange.com/questions/47749/how-secure-is-secure-keyboard-entry-in-mac-os-xs-terminal) and [here](https://rakhesh.com/mac/macos-find-app-using-secure-input/).

also, props to the sharemouse folks for showing that helpful warning.

~edit - I should mentioned that I think Evernote is the culprit, but I'll have to observe over a few days to verify.~

edit: [one more doc](https://smilesoftware.com/textexpander/secureinput) that's helpful 

cc @domparry 

---

> Commented on: 2019-09-17 by @domparry

@metadave, you're awesome. Spot on. My issue was secureinput locked by a startup app (workflowy). Huge Thank you. 

---

> Commented on: 2019-09-25 by @domparry

I'm closing this because it's solved (And not a Barrier issue)

---

> Commented on: 2020-11-23 by @iulo

@metadave thanks bro, save my life.
for other who face on this problem, a possiable solution is just: locking and unlocking your system.

---

> Commented on: 2020-12-04 by @royriojas

> @metadave thanks bro, save my life.
> for other who face on this problem, a possiable solution is just: locking and unlocking your system.

This is the simplest solution so far... many many thanks @iulo 

---

> Closed on: 2019-09-25 by @domparry

**Comment by tjuerges** [comment]
Just to add information for others who might run into this, too: In my case (macOS 12.1, Monterey) the process `/System/Library/CoreServices/loginwindow.app/Contents/MacOS/loginwindow console` was reported to use SecureInput. But it turned out that the real culprit was actually LastPass. Once I quit LastPass (Safari extension disabled, too), everything went back to normal.
